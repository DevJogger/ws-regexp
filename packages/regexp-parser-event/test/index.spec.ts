import { ParserEventEmitterEventList, ParserEventEmitterEvent, ParserEventEmitter } from '../index';

test(`ParserEventEmitterEventList`, () =>
{

	let actual = ParserEventEmitterEventList;
	// @ts-ignore
	let expected = Object.keys(ParserEventEmitterEvent);

	expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test('(^|[^\\w\'’])(\\d+(?:,\\d+)*)(?![\\w\'’])', () =>
{

	const str: string = '(^|[^\\w\'’])(\\d+(?:,\\d+)*)(?![\\w\'’])';

	let ev = ParserEventEmitter.create(str, 'u');
	ev.resume();

	let actual = ev.getSource(true);
	let expected = str;

	expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test(`麻#_@_#痹`, () =>
{

	const str: string = `麻#_@_#痹`;

	let ev = ParserEventEmitter.create(str, 'u');
	ev.resume();

	let actual = ev.getSource(true);
	let expected = str;

	expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});
