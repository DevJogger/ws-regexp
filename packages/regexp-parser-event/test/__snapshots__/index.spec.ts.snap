// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(^|[^\\w'’])(\\d+(?:,\\d+)*)(?![\\w'’]) 1`] = `"(^|[^\\\\w'’])(\\\\d+(?:,\\\\d+)*)(?![\\\\w'’])"`;

exports[`ParserEventEmitterEventList 1`] = `
Array [
  "default",
  "class",
  "other",
  "uniset",
  "class_default",
  "class_range",
  "class_other",
  "class_uniset",
  "change",
]
`;

exports[`麻#_@_#痹 1`] = `"麻#_@_#痹"`;
